skyrim-platform/src/platform_se/skyrim_platform/BrowserApi.cpp

```diff
@@ -89,3 +89,16 @@ JsValue BrowserApi::ExecuteJavaScript(const JsFunctionArguments& args,
   GetApp(state).ExecuteJavaScript(str);
   return JsValue::Undefined();
 }
+
+JsValue BrowserApi::UrlShellExecute(const JsFunctionArguments& args,
+                                    std::shared_ptr<State> state)
+{
+  auto str = static_cast<std::string>(args[1]);
+  if (str.substr(0, 8) != "https://") {
+    throw std::runtime_error(
+      "Permission denied, only 'https://' prefix is allowed");
+  } else {
+    ShellExecute(0, 0, str.c_str(), 0, 0, SW_SHOW);
```

Comment:
The program needs to call CoInitializeEx before it can be able to call ShellExecute.
BUT. If you can confirm that your code works, then it seems we have CoInitializeEx somewhere else in the program.

So. Let's just leave a comment that CoInitializeEx is needed for people that will support this code in the future. Like:

```c++
// CoInitializeEx is needed to call this function, but it seems we already have this call somewhere in code
```

we also better leave a docs link
https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecutea